pkg/cachemanager/cachemanager.go

```diff
@@ -318,6 +318,13 @@ func (c *CacheManager) syncGVK(ctx context.Context, gvk schema.GroupVersionKind)
 func (c *CacheManager) manageCache(ctx context.Context) {
 	// relistStopChan is used to stop any list operations still in progress
 	relistStopChan := make(chan struct{})
+	// waitToCloseChan is used to wait on the relist goroutine to end
+	// when needing to create another one. This ensures that we are essentially
+	// only using a singleton routine to relist gvks.
+	waitToCloseChan := make(chan struct{})
+
+	// the 0th relist goroutine is stopped, by definition, so we close the channel.
+	close(waitToCloseChan)
```

Comment:
Can you share why we are closing it right after it's created? I'm not clear on the comment.

Reply:
iirc, without closing the channel the first time around, the select block below will fire off an error (after 10 seconds hanging on a channel that is not signaled yet).

does that make sense? if so, how could we make the comment better?

Reply:
@ritazh let me know if the updated comment address the why of not closing the `relistStopChan` here. 🙏🏼